### PR TITLE
Drawing

### PR DESCRIPTION
**Pull request by jmatholia**
Moved some style related code from js to css. Created and added functionality to the modification mode, movement of nodes now allowed. Files modified CssGen.hs, DrawResponse.hs and experiment.js.

### CODE REVIEW

File: js/experiment.js [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
 
 function setupSVGCanvas() {
     'use-strict';
 
     var svg = document.createElementNS(xmlns, 'svg');
     svg.setAttribute('id', 'mySVG');
-    svg.setAttribute('style', 'border: 2px solid black');
-    svg.setAttribute('width', '800');
-    svg.setAttribute('height', '500');
+    svg.setAttribute('style', 'border: 2px solid black');   // also should go in CSS?
```
Comment (david-yz-liu) [inline js/experiment.js]: Yes, all node styling should be done in the CSS

File: js/experiment.js [hunk 2]
```diff
@@ -49,60 +54,131 @@ function getPosition(elem) {
...
 }
 
+
+function nodeMoved(e) {
+    'use-strict';
+
+    if (mode === 'change-mode' && nodeSelected !== null) {
+        nodeSelected = this;
+        var position = getClickPosition(e);
+        var rectX = e.currentTarget.getAttribute('x')*1;
```
Comment (david-yz-liu) [inline js/experiment.js]: Better to use `parseFloat` here

File: js/experiment.js [hunk 3]
```diff
@@ -49,60 +54,131 @@ function getPosition(elem) {
...
+
 function changeMode(id) {
-    //if (mode !== '') {
-    //  oldMode = document.getElementById(mode);
-    //  oldMode.setAttribute('border', 'border: 2px solid #008080');
-    //}
+    'use-strict';
+
+    if (mode !== '') {
+      $('#'+mode).toggleClass('clicked');
```
Comment (Ian-Stewart-Binks) [inline js/experiment.js]: Space around `+`, same for below.

File: js/experiment.js [hunk 4]
```diff
@@ -49,60 +54,131 @@ function getPosition(elem) {
...
+    }
     mode = id;
-    //newMode = document.getElementById(mode);
-    //newMode.setAttribute('border', 'border: 2px solid white');
+    newMode = document.getElementById(mode);
```
Comment (Ian-Stewart-Binks) [inline js/experiment.js]: I don't see `newMode` as a global variable, nor used anywhere else in this file. What is its purpose?

File: js/experiment.js [hunk 5]
```diff
@@ -49,60 +54,131 @@ function getPosition(elem) {
...
+    'use-strict';
+
+    oldColour = document.getElementById(nodeColourId);      // clicked class
+    oldColour.style.opacity = 1;
+    nodeColourId = id;
     nodeColour = colours[id];
-    //newMode = document.getElementById(mode);
-    //newMode.setAttribute('border', 'border: 2px solid white');
+    newColour = document.getElementById(nodeColourId);
+    newColour.style.opacity = 0.5;
```
Comment (Ian-Stewart-Binks) [inline js/experiment.js]: Where is `newColour` declared?

File: js/experiment.js [hunk 6]
```diff
@@ -49,60 +54,131 @@ function getPosition(elem) {
...
-    changeColour('blue');});
-$('#purple').click(function () {
-    changeColour('purple');});
-
 
 setupSVGCanvas();
-document.getElementById('mySVG').addEventListener('click', getClickPosition, false);
\ No newline at end of file
+
+$( '.mode' ).each(function( index ) {
```
Comment (Ian-Stewart-Binks) [inline js/experiment.js]: The spacing here looks unfamiliar.

**Comment by david-yz-liu** [comment]
@jmatholia Looking good, although the Firefox problem is likely caused by you using some function for checking for the mouse positions. The styling still needs to be moved over to CSS. The reason the CSS isn't having an impact is because in the actual graph, each rect (and text) is enclosed inside a <g> element, and this element is what actually has class .node, not the rect itself. You should make the same change in the drawing tool.

One thing I forgot in the meeting today: users should somehow be able to add text children to the nodes. The easiest way to do this would be to have a separate form with text input on the left side, and have the text child of a node update to that text value when you click a button or something.

And I'm looking forward to seeing the paths! I would prioritize both of these two over booleans and hybrids.

**Pull request merged**